bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java

```diff
@@ -0,0 +1,525 @@
+    }
+
+    private @Nullable State updateNumber(final String valueName) throws GreeException {
+        if (device.hasStatusValChanged(valueName)) {
+            return new DecimalType(device.getIntStatusVal(valueName));
+        }
+        return null;
+    }
+
+    private @Nullable State updateMode() throws GreeException {
+        if (device.hasStatusValChanged(GREE_PROP_MODE)) {
+            int mode = device.getIntStatusVal(GREE_PROP_MODE);
+            String modeStr = "";
+            switch (mode) {
+                case GREE_MODE_AUTO:
+                    modeStr = MODE_AUTO;
+                    break;
+                case GREE_MODE_COOL:
+                    boolean powerSave = device.getIntStatusVal(GREE_PROP_PWR_SAVING) == 1;
+                    modeStr = !powerSave ? MODE_COOL : MODE_ECO;
+                    break;
+                case GREE_MODE_DRY:
+                    modeStr = MODE_DRY;
+                    break;
+                case GREE_MODE_FAN:
+                    modeStr = MODE_FAN;
+                    break;
+                case GREE_MODE_HEAT:
+                    modeStr = MODE_HEAT;
+                    break;
+                default:
+                    modeStr = String.valueOf(mode);
+
+            }
+            if (!modeStr.isEmpty()) {
+                logger.debug("{}: Updading mode channel with {}/{}", thingId, mode, modeStr);
+                return new StringType(modeStr);
+            }
+        }
+        return null;
+    }
+
+    private @Nullable State updateQuiet() throws GreeException {
+        if (device.hasStatusValChanged(GREE_PROP_QUIET)) {
+            switch (device.getIntStatusVal(GREE_PROP_QUIET)) {
+                case GREE_QUIET_OFF:
+                    return new StringType(QUIET_OFF);
+                case GREE_QUIET_AUTO:
+                    return new StringType(QUIET_AUTO);
+                case GREE_QUIET_QUIET:
+                    return new StringType(QUIET_QUIET);
+            }
+        }
+        return null;
+    }
+
+    private @Nullable State updateTemp() throws GreeException {
+        if (device.hasStatusValChanged(GREE_PROP_SETTEMP) || device.hasStatusValChanged(GREE_PROP_TEMPUNIT)) {
+            int unit = device.getIntStatusVal(GREE_PROP_TEMPUNIT);
+            return toQuantityType(device.getIntStatusVal(GREE_PROP_SETTEMP), DIGITS_TEMP,
+                    unit == TEMP_UNIT_CELSIUS ? SIUnits.CELSIUS : ImperialUnits.FAHRENHEIT);
+        }
+        return null;
+    }
+
+    public static State toQuantityType(int value, int digits, Unit<?> unit) {
+        BigDecimal bd = new BigDecimal(value);
+        return new QuantityType<>(bd.setScale(digits, BigDecimal.ROUND_HALF_EVEN), unit);
+    }
+
+    public static QuantityType<?> toQuantityType(DecimalType value, int digits, Unit<?> unit) {
+        BigDecimal bd = new BigDecimal(value.doubleValue());
+        return new QuantityType<>(bd.setScale(digits, BigDecimal.ROUND_HALF_EVEN), unit);
+    }
```

Comment:
These can be merged together.
```suggestion
    public static QuantityType<?> toQuantityType(Number value, int digits, Unit<?> unit) {
        BigDecimal bd = new BigDecimal(value.doubleValue());
        return new QuantityType<>(bd.setScale(digits, BigDecimal.ROUND_HALF_EVEN), unit);
    }
```

Reply:
changed